packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2093,6 +2097,38 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     });
   }
 
+  async setPartitionConfig(
+    model: PartitionedGridModel,
+    rowIndex: GridRangeIndex
+  ): Promise<void> {
+    const keyTable = await this.pending.add(
+      model.partitionKeysTable(),
+      resolved => resolved.close()
+    );
+    assertNotNull(rowIndex);
+    keyTable.setViewport(rowIndex, rowIndex);
```

Comment:
You don't need to use the `keyTable` at all in this method, take it out. Just get the data from `this.getRowDataMap` like you're doing below, filter the values, and then set `partitionConfig`